spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkFilterUtilsTest.java

```diff
@@ -329,11 +329,11 @@ public void testFiltersWithNestedOrAnd_3() {
         pushAllFilters,
         dataFormat,
         "",
-        "(((((((`c1` >= 5000) OR (`c1` <= 701))) AND "
-            + "(((`c2` >= 150) OR (`c3` >= 100))))) OR (((((`c1` >= 50) OR "
-            + "(`c1` <= 71))) AND (((`c2` >= 15) OR (`c3` >= 10)))))) AND "
+        "((((`c1` >= 5000) OR (`c1` <= 701)) AND "
+            + "((`c2` >= 150) OR (`c3` >= 100))) OR (((`c1` >= 50) OR "
+            + "(`c1` <= 71)) AND ((`c2` >= 15) OR (`c3` >= 10)))) AND "
             + "((`c1` >= 500) OR (`c1` <= 70)) AND ((`c1` >= 900) OR "
-            + "(((`c3` <= 50) AND (((`c2` >= 20) OR (`c3` > 200)))))))",
+            + "((`c3` <= 50) AND ((`c2` >= 20) OR (`c3` > 200))))",
```

Comment:
Can you please add the test case where the generated SQL was incorrect before the fix?

Reply:
@vishalkarve15 yep, added the specific example I was describing.